chess/ChessGameModel.swift

```diff
@@ -137,30 +138,40 @@ extension ChessGameModel {
     }
 
     private func handle(_ message: String) {
-        guard isValidNotation(message) || isValidColor(message) else {
-            self.logger.error("Received invalid message: \(message)")
-            self.gameStatus = "Invalid move"
-            self.board = previousBoardState
-            self.player?.isMyTurn.toggle()
+        let messageType = findTypeOf(message)
+
+        guard let player = self.player, messageType != .color else {
```

Comment:
let's try to reduce complexity for this condition checks.
- i assume code within else condition can be a separate method?
- does it make sense to use `if` over `guard`?